flink-streaming-java/src/test/java/org/apache/flink/streaming/api/environment/CheckpointConfigFromConfigurationTest.java

```diff
@@ -90,7 +97,81 @@
                         .whenSetFromFile("execution.checkpointing.unaligned", "true")
                         .viaSetter(CheckpointConfig::enableUnalignedCheckpoints)
                         .getterVia(CheckpointConfig::isUnalignedCheckpointsEnabled)
-                        .nonDefaultValue(true));
+                        .nonDefaultValue(true),
+                TestSpec.testValue(
+                                (CheckpointStorage)
+                                        new FileSystemCheckpointStorage(
+                                                "file:///path/to/checkpoint/dir"))
+                        .whenSetFromFile("state.checkpoints.dir", "file:///path/to/checkpoint/dir")
+                        .viaSetter(CheckpointConfig::setCheckpointStorage)
+                        .getterVia(CheckpointConfig::getCheckpointStorage)
+                        .nonDefaultValue(
+                                new FileSystemCheckpointStorage("file:///path/to/checkpoint/dir"))
```

Comment:
shouldn't the paths be different then the configured path? As is, if the config doesn't forward it then surely the assertion would still succeed.

Reply:
There is no default value for `CheckpointStorage`. The `testNotOverridingIfNotSet` test verifies that the defined non-default value is still set after `configure` is called on an `Configuration` instance without any set values.

Reply:
I hate this test class already.